### PR TITLE
Added application generation support for the JDL based on JCore v2

### PR DESCRIPTION
**Pull request by MathieuAA**
- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [X] Tests are added where necessary
- [X] Documentation is added/updated where necessary
- [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

Notice the subtle bump from v1.4.6 to v2.0.1 ;)

### CODE REVIEW

**Comment by DanielFran** [comment]
@MathieuAA will you create a PR to update project JDL samples?

**Comment by MathieuAA** [comment]
@DanielFran yes :)

**Comment by MathieuAA** [comment]
@deepu105 I was wondering if the way to generate an app is as simple as generating an entity... From the code, it looks like it's really easy but I'd rather your opinion before committing the code.

**Comment by jdubois** [comment]
@MathieuAA have you also looked for upgrading https://github.com/jhipster/jdl-studio ?

**Comment by MathieuAA** [comment]
@jdubois Not yet

**Comment by deepu105** [comment]
@MathieuAA give me some time, i'll take a look mostly this weekend.

**Comment by deepu105** [comment]
> @deepu105 I was wondering if the way to generate an app is as simple as generating an entity... From the code, it looks like it's really easy but I'd rather have your opinion before committing the code.

@MathieuAA I didn't understand this comment. Your PR already creates the application from JDL right?

**Comment by MathieuAA** [comment]
@deepu105 it creates the required .yo-rc.json files and folders (and generates .jhipster + JSON entity files if needed). It doesn't call the app subgen, that was my question :)

**Comment by jdubois** [comment]
This is good for me -> @deepu105 let's wait until your review this week-end

**Comment by deepu105** [comment]
@MathieuAA thats easy in the import-jdl index.js new line 156 where we call `this.getExistingEntities()` just add this inside the try block
```
if (Object.keys(jdlObject.applications).length !== 0) {
     this.composeWith(require.resolve('../app'), {
           force: this.options.force,
           debug: this.options.debug,
           'skip-install': this.options['skip-install'],
           ... other flags that needs to be passed
     });
}
this.getExistingEntities().forEach((entity) => {
...
});
```

also, ensure to skip `this.rebuildClient();` in the end `if (Object.keys(jdlObject.applications).length !== 0)` to avoid double webpack build

**Comment by MathieuAA** [comment]
Thanks Deepu! I'll update the PR.

By the way, have you try fresh-installing JDL-studio lately? I've tried 
setting it up from scratch yesterday evening and it gave me headaches.

Some issues I've had may have been caused by not fixing the versions, 
but I'm not 100% sure yet. Will try to get it working right after.

**Comment by deepu105** [comment]
@MathieuAA I didn't do fresh install recently, i can try this weekend. Btw use NPM as it doesnt work well with YARN due to flat structuring

**Comment by MathieuAA** [comment]
Roger that!

File: generators/import-jdl/index.js [hunk 1]
```diff
@@ -92,77 +92,159 @@ module.exports = class extends BaseGenerator {
...
+                            entities,
+                            forceNoFiltering: this.options.force,
+                            application: {
+                                type: this.applicationType,
+                                name: this.baseName
+                            }
+                        });
                     }
-                } catch (e) {
-                    this.debug('Error:', e);
```
Comment (deepu105) [inline generators/import-jdl/index.js]: this was added to help errors be more understandable to end users, `this.error(`Error while parsing applications and entities from the JDL ${error}.`);` is not very helpful all the time, so can you retain these, please

File: generators/import-jdl/index.js [hunk 2]
```diff
@@ -92,77 +92,159 @@ module.exports = class extends BaseGenerator {
...
+                    this.error(`While generating applications and entities from the parsed JDL\n${error}`);
                 }
             }
         };
     }
 
     end() {
-        if (!this.options['skip-install'] && !this.skipClient && !this.options['json-only']) {
+        if (!this.options['skip-install'] && !this.skipClient && !this.options['json-only']
+                && Object.keys(this.jdlObject.applications).length === 0) {
```
Comment (deepu105) [inline generators/import-jdl/index.js]: this can be `!shouldGenerateApplications`

File: generators/import-jdl/index.js (hunk 1 above)
Comment (MathieuAA) [reply generators/import-jdl/index.js]: Will do

File: generators/import-jdl/index.js (hunk 2 above)
Comment (MathieuAA) [reply generators/import-jdl/index.js]: Well spotted!

**Comment by jdubois** [comment]
I generated several applications, including some pretty complex ones, everything looked perfect to me!
So I'm merging this -> @MathieuAA can you still correct later @deepu105 comments about `this.debug('Error:', e);`?

**Comment by MathieuAA** [comment]
@jdubois it is fixed, but as I've changed the code GH's diff didn't pick it up...

**Comment by DanielFran** [comment]
@MathieuAA , sorry to remind you but it is missing update/add jdl files in project JDL samples...

**Comment by MathieuAA** [comment]
You needn't be sorry, I completely forgot  :)At least the doc's ready.Le 30 mars 2018 19:00, Daniel Franco <notifications@github.com> a écrit :@MathieuAA , sorry to remind you but it is missing update/add jdl files in project JDL samples...

**Pull request merged**